pyrokinetics/local_geometry/LocalGeometryFourier.py

```diff
@@ -0,0 +1,662 @@
+
+        Parameters
+        ----------
+        eq : GlobalEquilibrium
+            GlobalEquilibrium object
+        psi_n : Float
+            Value of :math:`\psi_N` to generate local fourier parameters
+        verbose : Boolean
+            Controls verbosity
+
+        """
+
+        R, Z = eq.get_flux_surface(psi_n=psi_n)
+
+        b_poloidal = eq.get_b_poloidal(R, Z)
+
+        R_major = eq.R_major(psi_n)
+
+        rho = eq.rho(psi_n)
+
+        r_minor = rho * eq.a_minor
+
+        kappa = (max(Z) - min(Z)) / (2 * r_minor)
+
+        Zmid = (max(Z) + min(Z)) / 2
+
+        Zind = np.argmax(abs(Z))
+
+        R_upper = R[Zind]
+
+        fpsi = eq.f_psi(psi_n)
+        B0 = fpsi / R_major
+
+        drho_dpsi = eq.rho.derivative()(psi_n)
+        shift = eq.R_major.derivative()(psi_n) / drho_dpsi / eq.a_minor
+        dZ0dr = eq.Z_mid.derivative()(psi_n) / drho_dpsi / eq.a_minor
+
+        pressure = eq.pressure(psi_n)
+        q = eq.q(psi_n)
+
+        dp_dpsi = eq.q.derivative()(psi_n)
+
+        shat = rho / q * dp_dpsi / drho_dpsi
+
+        dpressure_drho = eq.p_prime(psi_n) / drho_dpsi
+
+        beta_prime = 8 * pi * 1e-7 * dpressure_drho / B0**2
+
+        normalised_height = (Z - Zmid) / (kappa * r_minor)
+
+        # Floating point error can lead to >|1.0|
+        normalised_height = np.where(
+            np.isclose(normalised_height, 1.0), 1.0, normalised_height
+        )
+        normalised_height = np.where(
+            np.isclose(normalised_height, -1.0), -1.0, normalised_height
+        )
+
+        theta = np.arcsin(normalised_height)
+
+        normalised_radius = (R - R_major) / r_minor
+
+        normalised_radius = np.where(
+            np.isclose(normalised_radius, 1.0), 1.0, normalised_radius
+        )
+        normalised_radius = np.where(
+            np.isclose(normalised_radius, -1.0), -1.0, normalised_radius
+        )
+
+        thetaR = np.arccos(normalised_radius)
+
+        for i in range(len(theta)):
+            if R[i] < R_upper:
+                theta[i] = np.pi - theta[i]
+            else:
+                if Z[i] < 0:
+                    theta[i] = 2 * np.pi + theta[i]
+
+        for i in range(len(thetaR)):
+            if Z[i] < 0:
+                thetaR[i] = 2 * np.pi - thetaR[i]
+
+        theta_diff = thetaR - theta
+
+        self.psi_n = psi_n
+        self.rho = float(rho)
+        self.r_minor = float(r_minor)
+        self.Rmaj = float(R_major / eq.a_minor)
+        self.a_minor = float(eq.a_minor)
+        self.f_psi = float(fpsi)
+        self.B0 = float(B0)
+
+        self.Z0 = float(Zmid / eq.a_minor)
+        self.R = R
+        self.Z = Z
+        self.theta = theta
+        self.b_poloidal = b_poloidal
+
+        self.q = float(q)
+        self.shat = shat
+        self.beta_prime = beta_prime
+        self.pressure = pressure
+        self.dpressure_drho = dpressure_drho
+
+        for n_moments in [4, 5, 6, 8, 10, 12, 14, 16]:
```

Comment:
Is this something we might want to customise? Or is this a fairly standard set of moments?

Reply:
We can put it as a `kwargs` You only need the one but that is a choice. My only concern is that if codes only take 3 moments then we'll need to fit 3 moments. Not sure if you can just truncate a fourier series. There may be some thinking required here